### PR TITLE
Integration with strapi

### PR DESCRIPTION
**Pull request by komalahire**
Merd Integration

 Worked on Header, Footer, Student dashboard, and home page, removed hard-coded pathway data from all components, and Integrate with API

@jschanker

### CODE REVIEW

File: src/components/PathwayCourse/index.js [hunk 1]
```diff
@@ -132,12 +131,16 @@ function PathwayCourse() {
   const [openSnackbar, setOpenSnackbar] = useState(false);
   const [openModal, setOpenModal] = useState(false);
   const [certificate, setCertificate] = useState("");
-  let completedAll = pathwayId==8?completedPortion?.total>=80: completedPortion?.total === 100;
+  let completedAll =
+    pathwayId == 8
+      ? completedPortion?.total >= 80
+      : completedPortion?.total === 100;
```
Comment (jschanker) [inline src/components/PathwayCourse/index.js]: Few comments here:
1. Using `==` will add lint. Maybe use `parseInt(pathwayId) === 8` instead.
2. We don't want to be hardcoding ids or minimum completed percentages here as they can change. We could remove the reference to the id here of `8` and the minimum completion percentage of `80` if we instead have an object mapping ids to the values necessary to complete them. If the id isn't present, we could use `100` (e.g., `const MINIMUM_COMPLETE_PERCENTAGES = {'8': 80}`) . This would allow us to easily change minimum completed portions of other pathways as well without touching the code. However, it would be better to get this map of completed percentages from the back-end. For now (after importing this constant):
```suggestion
    completedPortion?.total >= (MINIMUM_COMPLETE_PERCENTAGES[pathwayId] || 100);
```

File: src/components/PathwayCourse/index.js [hunk 2]
```diff
@@ -334,8 +339,7 @@ function PathwayCourse() {
                 align="center"
                 className={classes.classTitle}
                 variant="subtitle1"
-                bgcolor="#E9F5E9"
-              >
+                bgcolor="#E9F5E9">
```
Comment (jschanker) [inline src/components/PathwayCourse/index.js]: Reference theme color here?

File: src/components/PathwayCourse/index.js [hunk 3]
```diff
@@ -132,12 +131,16 @@ function PathwayCourse() {
   const [openSnackbar, setOpenSnackbar] = useState(false);
   const [openModal, setOpenModal] = useState(false);
   const [certificate, setCertificate] = useState("");
-  let completedAll = pathwayId==8?completedPortion?.total>=80: completedPortion?.total === 100;
+  let completedAll =
```
Comment (jschanker) [inline src/components/PathwayCourse/index.js]: `completedAll` is a little misleading as a variable name if you only need to complete a minimum percentage. Also, use `const` here since the variable is not reassigned.

**Review by jschanker — COMMENTED** [review]
Some initial comments. Will continue reviewing later.

**Comment by jschanker** [comment]
Issue with the dropdown image sizes?
image



What is your screen size @jschanker

File: src/components/NewUserDashbord/index.js [hunk 4]
```diff
@@ -44,22 +44,13 @@ const NewUserDashbord = () => {
...
-        pathwayData.push(obj);
-      }
-    });
+  const MiscellaneousCourses = data?.pathways.filter(
+    (item) => item.name === "Miscellaneous Courses"
+  );
+
+  const pathwayData = data?.pathways
+    .filter((item) => item.name !== "Miscellaneous Courses")
+    .concat(MiscellaneousCourses);
```
Comment (jschanker) [inline src/components/NewUserDashbord/index.js]: Is the reason for this code to include the Miscellaneous courses pathway at the end? We'll probably want to avoid explicit mention of the name in case we decide to change its name or offer translations into other languages. Perhaps we could `import` and use `PATHWAY_INFO` to determine whether it should be at the end of not. Also, we should probably rename the variable to indicate that it's a pathway name and not the collection of miscellaneous courses such as Web Scraping 101.

For example:
```suggestion
  const miscellaneousPathway = data?.pathways.filter(
    (pathway) => PATHWAY_INFO.some((miscPathway) => pathway.name === miscPathway.name)
  );

  const pathwayData = data?.pathways
    .filter((pathway) => !miscellaneousPathway.includes(pathway))
    .concat(miscellaneousPathway);
```

File: src/pages/Home/index.js [hunk 5]
```diff
@@ -161,23 +161,9 @@ function Home() {
...
-          image: pathway.image || pathway.logo,
-          link: pathway.path || null,
-          type: "internal",
-        };
-        pathwayData.push(obj);
-      }
-    });
+  const MiscellaneousCourses = data?.pathways.filter(item => item.name === 'Miscellaneous Courses')
 
+  const pathwayData = data?.pathways.filter(item => item.name !== 'Miscellaneous Courses').concat(MiscellaneousCourses)
```
Comment (jschanker) [inline src/pages/Home/index.js]: Similar comment here about explicit mention of "Miscellaneous courses."

File: src/components/PathwayCourse/index.js [hunk 6]
```diff
@@ -132,12 +131,16 @@ function PathwayCourse() {
...
+    pathwayId == 8
+      ? completedPortion?.total >= 80
+      : completedPortion?.total === 100;
   // let completedAll = true
   const [loader, setLoader] = useState(false);
-  const displayCert = pathwayId == 1 
+  const displayCert = pathwayId == 1;
```
Comment (jschanker) [inline src/components/PathwayCourse/index.js]: Similar comment to https://github.com/navgurukul/bhanwari-devi/pull/1002/files#r1229700374. There's the lint issue of using `==` and also we want to avoid explicit references to pathway ids in case they change. So, let's maybe add a line `const IDS_OF_PATHWAYS_WITH_CERTIFICATES = ['1'];` to the constants file and then we could do the following. This would also have the added benefit of allowing us to add pathways with certificates in the future by merely updating the constants file, without touching the code. Again, it'd be good to get this information from the back-end at some point.
```suggestion
  const displayCert = IDS_OF_PATHWAYS_WITH_CERTIFICATES.includes(pathwayId);
```

File: src/components/PathwayCourse/index.js [hunk 7]
```diff
@@ -395,29 +397,28 @@ function PathwayCourse() {
...
-                      >
+                        mb={isActive ? 16 : 30}>
                         {pathwayCourseData.code !== "SHCEL" && (
                           <Typography
                             variant="body2"
                             className={classes.cardSubtitle}
-                            sx={{ textAlign: isActive && "center", pb: "8px" }}
-                          >
+                            sx={{
+                              textAlign: isActive && "center",
```
Comment (jschanker) [inline src/components/PathwayCourse/index.js]: Not really part of this PR, but it would be good to remove `isActive` as possible to have the styles update with CSS media queries instead of re-renders.
```suggestion
                              textAlign: {xs: "center", sm: "left"},
```

File: src/pages/Home/PathwayCard.js [hunk 8]
```diff
@@ -4,45 +4,40 @@ import { Link } from "react-router-dom";
...
+          {logo && (
             <CardMedia
               component="img"
               src={
-                image.includes('https')
-                  ? image
-                  : require("./assets/" + image + ".svg")
+                logo.includes("https")
+                  ? logo
+                  : require("./assets/" + logo + ".svg")
```
Comment (jschanker) [inline src/pages/Home/PathwayCard.js:29]: May want to have a function that returns the paths of the logos to use based on what comes from the back-end. This would allow for changes (e.g., if logos aren't going to be in assets folder) without editing the logic in multiple components.

**Comment by Poonam-Singh-Bagh** [comment]
> Issue with the dropdown image sizes? image

File: src/components/PathwayCourse/index.js (hunk 1 above)
Comment (Poonam-Singh-Bagh) [reply src/components/PathwayCourse/index.js]: For now, we are going for `pathway.code`. After it's released it will come from the backend

File: src/components/NewUserDashbord/index.js (hunk 4 above)
Comment (komalahire) [reply src/components/NewUserDashbord/index.js]: Okay, I will change it but I will ask the backend to send this course from the backend only.

**Comment by jschanker** [comment]
> Issue with the dropdown image sizes? image
> 
> What is your screen size @jschanker

@Poonam-Singh-Bagh Screen or browser window width? It happens at any browser window width for me. This seems to be due to some of the images being larger than the ones that they're replacing. For example, the Typing one located at https://merakilearn.s3.ap-south-1.amazonaws.com/Meraki+track+icons/typing/type%3Dtyping_xxxhdpi.png is 128 x 128 as opposed to the Python one at https://merakilearn.s3.ap-south-1.amazonaws.com/Meraki+track+icons/python/Python.png, which is only 48 x 48.
